internal/controllers/gateway/gateway_utils.go

```diff
@@ -297,11 +298,29 @@ func getListenerStatus(
 					if !apierrors.IsNotFound(err) {
 						return nil, err
 					}
-					tlsresolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
+					tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
+					break
+				}
+				var ok bool
+				var crt, key []byte
+				if crt, ok = secret.Data["tls.crt"]; !ok {
+					tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
+					break
+				}
+				if key, ok = secret.Data["tls.key"]; !ok {
+					tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
+					break
+				}
+				if p, _ := pem.Decode(crt); p == nil {
+					tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
+					break
+				}
+				if p, _ := pem.Decode(key); p == nil {
+					tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
```

Comment:
Could we extract a separate function validating the cert secret? The usage could be as follows:

```go
ok, err := validateTLSCertSecret(ctx, client, certRef)
if err != nil {
  return nil, err 
}
if !ok {
  tlsResolvedRefReason = string(gatewayv1beta1.ListenerReasonInvalidCertificateRef)
  break
}
```